frontend/tests/unit/store/modules/configForm.spec.js

```diff
@@ -86,5 +108,82 @@ describe('configForm.js', () => {
     store.commit('setCommonServiceType', 'webade');
     expect(store.getters.usingWebadeConfig).toBe(true);
   });
+});
+
+describe('configForm.js - WebADE configs', () => {
+  let store;
+
+  const configBlank = require('../fixtures/webAdeConfigs/configBlank.json');
+  const configWithCmsg = require('../fixtures/webAdeConfigs/configWithCmsg.json');
+  const configWithNoServices = require('../fixtures/webAdeConfigs/configWithNoServices.json');
+  const configWithNrosDMS = require('../fixtures/webAdeConfigs/configWithNrosDMS.json');
+
+  beforeEach(() => {
+    const localVue = createLocalVue();
+    localVue.use(Vuex);
+
+    store = new Vuex.Store(cloneDeep(configFormStore));
+  });
+
+  it('returns the expected WebADE App config based on the user entered details - totally blank', () => {
+    const config = store.getters.appConfigAsString;
+    expect(config).toBeTruthy();
+    expect(config).toEqual(JSON.stringify(configBlank, null, 2));
+  });
+
+  it('returns the expected WebADE App config based on the user entered details - no services', () => {
+    store.state.userAppCfg = {
+      applicationAcronym: 'WORG',
+      applicationName: 'Document Generation',
+      applicationDescription: 'A sample application description test',
+      commonServices: []
+    };
+
+    const config = store.getters.appConfigAsString;
+    expect(config).toBeTruthy();
+    expect(config).toEqual(JSON.stringify(configWithNoServices, null, 2));
+  });
+
+  it('returns the expected WebADE App config based on the user entered details - cmsg', () => {
+    store.state.userAppCfg = {
+      applicationAcronym: 'WORG',
+      applicationName: 'Document Generation',
+      applicationDescription: 'A sample application description test',
+      commonServices: ['cmsg']
+    };
+
+    const config = store.getters.appConfigAsString;
+    expect(config).toBeTruthy();
+    expect(config).toEqual(JSON.stringify(configWithCmsg, null, 2));
+  });
+
+  it('returns the expected WebADE App config based on the user entered details - nros-dms', () => {
+    store.state.userAppCfg = {
+      applicationAcronym: 'MDS',
+      applicationName: 'Mines Digital Services',
+      applicationDescription: 'The technologies that support mine oversight',
+      commonServices: ['nros-dms']
+    };
+
+    const config = store.getters.appConfigAsString;
+    expect(config).toBeTruthy();
+    expect(config).toEqual(JSON.stringify(configWithNrosDMS, null, 2));
+  });
+
+  it('returns the expected WebADE App config with a placeholder secret for the manual deployment mode', () => {
+    store.state.userAppCfg = {
+      applicationAcronym: 'WORG',
+      applicationName: 'Document Generation',
+      applicationDescription: 'A sample application description test',
+      deploymentMethod: 'deploymentManual'
+    };
+
+    const configWithNoServicesCopy = JSON.parse(JSON.stringify(configWithNoServices));
```

Comment:
I presume you're doing parse/stringify to do a "deep" copy. As of Node v11 onwards, this can be achieved with the following direct v8 call:
```
const v8 = require('v8');

const structuredClone = obj => {
  return v8.deserialize(v8.serialize(obj));
};
```
Source: https://stackoverflow.com/a/10916838